### PR TITLE
Review public api spanmetrics

### LINKED ISSUE #4090: Review public API for `spanmetricsprocessor`
Relates to #4055

### PR DESCRIPTION
**Pull request by albertteoh**
**Description:** Hides exported structs that are not used by external packages.

**Link to tracking Issue:** Fixes #4090

**Testing:** 
- Unit tests passing
- Local integration tests show collector with spanmetrics processor continues to run as expected, and able to successfully load configuration.

### CODE REVIEW

File: processor/spanmetricsprocessor/config.go [hunk 1]
```diff
@@ -20,14 +20,14 @@ import (
...
+// dimension defines the dimension name and optional default value if the dimension is missing from a span attribute.
+type dimension struct {
 	Name    string  `mapstructure:"name"`
 	Default *string `mapstructure:"default"`
 }
 
-// Config defines the configuration options for spanmetricsprocessor.
-type Config struct {
+// spanMetricsConfig defines the configuration options for spanmetricsprocessor.
+type spanMetricsConfig struct {
```
Comment (albertteoh) [inline processor/spanmetricsprocessor/config.go:30]: I wasn't sure if we'd want to export `Config` for godoc generation as I've already documented configuration in the README. But if the convention is to document configs through godoc, happy to just close this PR and leave things as is.

Comment (albertteoh) [reply processor/spanmetricsprocessor/config.go:30]: For reference, here is the current godoc for spanmetricsprocessor: https://pkg.go.dev/github.com/open-telemetry/opentelemetry-collector-contrib/processor/spanmetricsprocessor

Comment (jpkrohling) [reply processor/spanmetricsprocessor/config.go:30]: I do not know the answer to this. @bogdandrutu?

**Review by jpkrohling — COMMENTED** [review]
LGTM, but needs clarification on whether reducing the visibility of the `Config` should be done.

File: processor/spanmetricsprocessor/config.go (hunk 1 above)
Comment (bogdandrutu) [reply processor/spanmetricsprocessor/config.go:30]: I also don't know about this. We should probably file an issue to discuss with the community if this is needed.

Comment (albertteoh) [reply processor/spanmetricsprocessor/config.go:30]: Yeah, when creating this processor, I was following the `routingprocessor`'s lead in exporting the `Config`.

Comment (jpkrohling) [reply processor/spanmetricsprocessor/config.go:30]: I ended up having Config public for other components, which is what makes the most sense to me.

Comment (albertteoh) [reply processor/spanmetricsprocessor/config.go:30]: Thanks @jpkrohling, in that case I'll close this PR and corresponding issue.

**Pull request closed without merging**